src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -4,11 +4,31 @@
 #---------------------------------------------------------------------------------------------
 #pylint: disable=line-too-long
 import azure.cli.commands.parameters #pylint: disable=unused-import
-
+from azure.cli.commands import CliArgumentType
 from azure.cli.commands import register_cli_argument
 
 register_cli_argument('ad app', 'application_object_id', options_list=('--object-id',))
-register_cli_argument('ad app', 'app_id', help='application id')
+register_cli_argument('ad app', 'display_name', help=' the display name of the application')
+register_cli_argument('ad app', 'homepage', help='the url where users can sign in and use your app.')
+register_cli_argument('ad app', 'identifier', options_list=('--id',), help='identifier uri, application id, or object id')
+register_cli_argument('ad app', 'identifier_uris', nargs='+', help='space separated unique URIs that Azure AD can use for this app.')
+register_cli_argument('ad app', 'reply_urls', nargs='+',
+                      help='space separated URIs to which Azure AD will redirect in response to an OAuth 2.0 request. The value does not need to be a physical endpoint, but must be a valid URI.')
+register_cli_argument('ad app', 'start_date', help='the start date after which password or key would be valid. Default value is current time')
+register_cli_argument('ad app', 'end_date', help='the end date till which password or key is valid. Default value is one year after current time')
+register_cli_argument('ad app', 'key_value', help='the value for the key credentials associated with the application')
+register_cli_argument('ad app', 'key_type', choices=['AsymmetricX509Cert', 'Password', 'Symmetric'], default='AsymmetricX509Cert',
```

Comment:
I thought about doing you suggested, but as i checked out, the value is not modeled in SDK.